src/config.rs

```diff
@@ -1850,16 +1931,19 @@ impl ConfigHandler {
             StatusCode::NotChange => Ok(()),
             StatusCode::WrongVersion if cmp_version(&self.version, &version) == Ordering::Less => {
                 let incoming: TiKvConfig = toml::from_str(resp.get_config())?;
-                if let Some(rollback_change) = self.config_controller.update_or_rollback(incoming) {
-                    debug!(
-                        "tried to update local config to an invalid config";
-                        "version" => ?resp.version
-                    );
-                    let entries = to_config_entry(rollback_change)?;
-                    self.update_config(version, entries, pd_client)?;
-                } else {
-                    info!("local config updated"; "version" => ?resp.version);
-                    self.version = version;
+                match self.config_controller.update_or_rollback(incoming)? {
+                    Some(rollback_change) => {
+                        debug!(
+                            "tried to update local config to an invalid config";
+                            "version" => ?resp.version
+                        );
+                        let entries = to_config_entry(rollback_change)?;
+                        self.update_config(version, entries, pd_client)?;
+                    }
+                    None => {
+                        info!("local config updated"; "version" => ?resp.version);
```

Comment:
If there is no update, it will also return None.

Reply:
Yes, you are right. Because we first compare our version with resp's version, so this can only happen when some 
 config that not support change had updated on pd side (so resp's version are higher but `diff` is empty), in this situation we need to update our version but not print the log. PTAL again